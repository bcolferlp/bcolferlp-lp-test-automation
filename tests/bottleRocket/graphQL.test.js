import moment from 'moment';
import GraphQL from '../../src/apis/graphQLAPI';
import Cognito from '../../src/apis/cognitoAPI';

require('dotenv').config();

jest.setTimeout(60000 * 5);

describe('GraphQL', () => {
  let token;
  let graphQL;
  const endPoint = 'https://br1-partner.api.loanpal.com/graphql';
  const config = {
    UserPoolId: process.env.cognitoUserPoolId,
    ClientId: process.env.cognitoClientId,
    Username: process.env.testEmail,
    Password: process.env.emailPass
  };

  beforeAll(async () => {
    const cog = new Cognito(config);
    token = await cog.getToken();
    graphQL = new GraphQL(endPoint, token);
  });

  test('97460: LPLBACK-8 Get terms and Conditions', async () => {
    const query = `{
        terms(current:true){
            id
            version
            sections{
              sort
              title
              subTitle
              text
            }
            current
          }
      }`;

    const result = await graphQL.request(query);
    expect(result).toMatchSnapshot();
  });

  test('97461: LPLBACK-9 Accept Terms and Conditions', async () => {
    const query = `
    mutation acceptTerms($termId: ID!){
        acceptTerms(termsId: $termId){
          id
          acceptedTerms{
            termsAndConditionsId
            dateAccepted
          }
        }
      }`;
    const date = moment().format('YYYY-MM-DD');
    const variable = { termId: 'TRMS-ddd9dec5-1ea8-46ef-b103-7ff74bd66456' };
    const result = await graphQL.request(query, variable);
    const allTerms = result.acceptTerms.acceptedTerms;
    const latestTerm = allTerms[allTerms.length - 1];
    console.log('latestTerm', latestTerm);
    expect(latestTerm).toEqual(expect.objectContaining({ termsAndConditionsId: variable.termId, dateAccepted: expect.stringMatching(date) }));
  });
});
