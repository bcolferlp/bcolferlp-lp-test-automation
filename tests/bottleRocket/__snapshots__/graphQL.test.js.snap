// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphQL 97460: LPLBACK-8 Get terms and Conditions 1`] = `
Object {
  "terms": Array [
    Object {
      "current": true,
      "id": "TRMS-ddd9dec5-1ea8-46ef-b103-7ff74bd66456",
      "sections": Array [
        Object {
          "sort": 6,
          "subTitle": null,
          "text": "The Services are provided “As Is” and “As Available”.  Loanpal disclaims all representations and warranties, express, implied, or statutory not expressly set out in there terms, including the implied warranties of merchantability, fitness for a particular purpose and non-infringement. ",
          "title": null,
        },
        Object {
          "sort": 5,
          "subTitle": null,
          "text": "You agree that Loanpal may contact you by email, phone or text messages at any of the contact number/addresses you provided.",
          "title": null,
        },
        Object {
          "sort": 4,
          "subTitle": null,
          "text": "The Services are provided pursuant to the terms of the Partner’s Solar/Storage Financing Program Agreement which terms are incorporated herein by reference.",
          "title": null,
        },
        Object {
          "sort": 3,
          "subTitle": null,
          "text": "In order to use the Services you must register for and maintain an active “User Account”.  You must be at least 18 years of age.  You may not authorize or transfer your User Account to any other person or entity. You are responsible for obtaining the data network access necessary to use the Services.",
          "title": null,
        },
        Object {
          "sort": 2,
          "subTitle": null,
          "text": "The information provided in the Services are and shall remain the property of Loanpal.",
          "title": null,
        },
        Object {
          "sort": 1,
          "subTitle": null,
          "text": "You must allow the applicant to provide their own information without undue pressure or influence and in a manner that preserves the integrity and confidentiality of the information that they have provided",
          "title": null,
        },
        Object {
          "sort": 0,
          "subTitle": null,
          "text": "You must not request or nor accept information which you reasonably believe to be false or misleading,",
          "title": null,
        },
      ],
      "version": "1.0.0",
    },
  ],
}
`;
